src/routes/safe/components/Settings/SpendingLimit/BeneficiarySelect.tsx

```diff
@@ -0,0 +1,109 @@
+import { EthHashInfo } from '@gnosis.pm/safe-react-components'
+import React from 'react'
+import { useForm, useFormState } from 'react-final-form'
+import { useSelector } from 'react-redux'
+import styled from 'styled-components'
+
+import { ScanQRWrapper } from 'src/components/ScanQRModal/ScanQRWrapper'
+import { getNetwork } from 'src/config'
+import { getAddressBook } from 'src/logic/addressBook/store/selectors'
+import { getNameFromAdbk } from 'src/logic/addressBook/utils'
+import AddressBookInput from 'src/routes/safe/components/Balances/SendModal/screens/AddressBookInput'
+
+const KEYCODES = {
```

Comment:
this could be used in another placer, perhaps moving it to a generic place is a good idea.

Reply:
Ok, I'll take this suggestion to move all the generic functions to a utils file, but inside the SpendingLimit/ directory for now